tutorials/large-messages/index.mdx

```diff
@@ -0,0 +1,377 @@
+    }
+
+    provider "scaleway" {
+      zone = "fr-par-1"
+    }
+    ```
+
+3. Create a `terraform.tfvars` and add your Scaleway access key and access key id.
+    ```terraform
+    scw_access_key_id     = "YOUR_SCW_SECRET_KEY_ID"
+    scw_secret_access_key = "YOUR_SCW_SECRET_KEY"
+    ```
+
+4. Still in `main.tf` add this terraform code to create a Object Storage bucket that you will use to store your images.
+    ```terraform
+    resource "random_id" "bucket" {
+      byte_length = 8
+    }
+
+    resource "scaleway_object_bucket" "large_messages" {
+      name = "large-messages-${random_id.bucket.hex}"
+    }
+
+    resource "scaleway_object_bucket_acl" "large_messages" {
+      bucket = scaleway_object_bucket.large_messages.id
+      acl    = "private"
+    }
+
+    output "bucket_name" {
+      value       = scaleway_object_bucket.large_messages.name
+      description = "Bucket name to use with the producer script"
+    }
+    ```
+    In this code, the ressource `random_id.bucket` create a random id, that you will pass to object bucket just after to be sure that the bucket name is unique.
+    An `scaleway_object_bucket_acl` is added on the bucket to make it private and output the bucket name to use it in your producer.
+
+5. Next, add these ressources to create a NATS account an your NATS credential file:
+    ```terraform
+    resource "scaleway_mnq_nats_account" "large_messages" {
+      name = "nats-acc-large-messages"
+    }
+
+    resource "scaleway_mnq_nats_credentials" "large_messages" {
+      name       = "nats-large-messages-creds"
+      account_id = scaleway_mnq_nats_account.large_messages.id
+    }
+
+    resource "local_file" "nats_credential" {
+      content         = scaleway_mnq_nats_credentials.large_messages.file
+      filename        = "large-messages.creds"
+      file_permission = 644
+    }
+
+    output "nats_url" {
+      value       = scaleway_mnq_nats_account.large_messages.endpoint
+      description = "NATS url to use with the producer script"
+    }
+    ```
+    We also output the NATS server url.
+
+6. You can run `terraform init` and `terraform apply` to create the ressources in your Scaleway account. 
+
+## Create the producer
+
+As said before, the producer will be a simple shell script.
+
+1. Create a file `upload_img.sh` 
+    In it create two variables that use terraform command to get bucket name and NATS Url:
+    ```bash
+    SCW_BUCKET="$(terraform output bucket_name)"
+    SCW_NATS_URL="$(terraform output nats_url)"
+    ```
+
+2. Next configure the NATS cli with the NATS url and the credentials file.
+    ```bash
+    nats context save large-messages --server=$SCW_NATS_URL --creds=./large-messages.creds
+    nats context select large-messages
+    ```
+    Our script take the file path that we want to upload as first parameter.
+
+3. To upload our file we gonna use the AWS cli configured with Scaleway endpoint and credentials because the Scaleway Object storage is fully compliant with S3.
```

Comment:
```suggestion
3. To upload our file we gonna use the AWS CLI configured with Scaleway endpoint and credentials because the Scaleway Object storage is fully compliant with S3.
```

Reply:
```suggestion
To upload the file, we will use the AWS CLI configured with the Scaleway endpoint and credentials, because the Scaleway Object storage is fully compliant with S3.
```